nautilus_trader/persistence/streaming/batching.py

```diff
@@ -79,28 +82,24 @@ def _generate_batches_rust(
     cls: type,
     batch_size: int = 10_000,
 ) -> Generator[list[Union[QuoteTick, TradeTick]], None, None]:
+    files = sorted(files, key=lambda x: Path(x).stem)
+
     assert cls in (QuoteTick, TradeTick)
 
-    # TODO: Replace with new Rust datafusion backend
-    yield []
-    # files = sorted(files, key=lambda x: Path(x).stem)
-    # for file in files:
-    #     reader = ParquetReader(
-    #         file,
-    #         batch_size,
-    #         py_type_to_parquet_type(cls),
-    #         ParquetReaderType.File,
-    #     )
-    #     for capsule in reader:
-    #         # PyCapsule > List
-    #         if cls == QuoteTick:
-    #             objs = QuoteTick.list_from_capsule(capsule)
-    #         elif cls == TradeTick:
-    #             objs = TradeTick.list_from_capsule(capsule)
-    #         else:
-    #             raise RuntimeError(f"Data type {cls} unsupported for Rust.")
-    #
-    #         yield objs
+    session = DataBackendSession(chunk_size=batch_size)
+
+    for file in files:
+        session.add_file_with_query(
+            "data",
+            file,
+            "SELECT * FROM data;",
```

Comment:
Session expects the data to be sorted by `ts_init`. It's ok to use this if you know the data is already sorted. If that's not the case you can use the `add_file_with_default_query` which internally calls `SELECT * FROM {} ORDER BY ts_init`.

Reply:
Thanks for checking, I've updated to use the new method now

https://github.com/nautechsystems/nautilus_trader/pull/1117/commits/7acf97e4f87607ce4d6abdd8f9f1a1faaccef3d3